core/src/main/kotlin/com/justai/jaicf/activator/strategy/ActivationByContextPenalty.kt

```diff
@@ -0,0 +1,115 @@
+package com.justai.jaicf.activator.strategy
+
+import com.justai.jaicf.context.BotContext
+import com.justai.jaicf.helpers.logging.WithLogger
+import com.justai.jaicf.model.activation.Activation
+import com.justai.jaicf.model.activation.ActivationStrategy
+import com.justai.jaicf.model.state.StatePath
+
+/**
+ * Implementation of [ActivationStrategy] by analogy with JAICP DSL.
+ *
+ * This activation strategy calculates penalty for each activation,
+ * applies penalties for every activation and selects single activation with highest adjusted confidence.
+ *
+ * Penalty increases with difference level between current state and activation target state.
```

Comment:
**Should be fixed**
Describe more properly what is the difference between states